File: sources/AtField 1.3/net/ccbluex/liquidbounce/utils/EntityUtils.java
/*
 * Decompiled with CFR 0.152.
 */
package net.ccbluex.liquidbounce.utils;

import net.ccbluex.liquidbounce.LiquidBounce;
import net.ccbluex.liquidbounce.api.minecraft.client.entity.IEntity;
import net.ccbluex.liquidbounce.api.minecraft.client.entity.player.IEntityPlayer;
import net.ccbluex.liquidbounce.api.minecraft.client.network.INetworkPlayerInfo;
import net.ccbluex.liquidbounce.api.minecraft.scoreboard.ITeam;
import net.ccbluex.liquidbounce.features.module.modules.combat.NoFriends;
import net.ccbluex.liquidbounce.features.module.modules.misc.AntiBot;
import net.ccbluex.liquidbounce.features.module.modules.misc.Teams;
import net.ccbluex.liquidbounce.utils.MinecraftInstance;

public final class EntityUtils
extends MinecraftInstance {
    public static boolean targetAnimals;
    public static boolean targetMobs;
    public static boolean targetDead;
    public static boolean targetInvisible;
    public static boolean targetPlayer;

    public static String getName(INetworkPlayerInfo iNetworkPlayerInfo) {
        if (iNetworkPlayerInfo.getDisplayName() != null) {
            return iNetworkPlayerInfo.getDisplayName().getFormattedText();
        }
        ITeam iTeam = iNetworkPlayerInfo.getPlayerTeam();
        String string = iNetworkPlayerInfo.getGameProfile().getName();
        return iTeam == null ? string : iTeam.formatString(string);
    }

    static {
        targetInvisible = false;
        targetPlayer = true;
        targetMobs = true;
        targetAnimals = false;
        targetDead = false;
    }

    public static int getPing(IEntityPlayer iEntityPlayer) {
        if (iEntityPlayer == null) {
            return 0;
        }
        INetworkPlayerInfo iNetworkPlayerInfo = mc.getNetHandler().getPlayerInfo(iEntityPlayer.getUniqueID());
        return iNetworkPlayerInfo == null ? 0 : iNetworkPlayerInfo.getResponseTime();
    }

    public static boolean isSelected(IEntity iEntity, boolean bl) {
        if (classProvider.isEntityLivingBase(iEntity) && (targetDead || iEntity.isEntityAlive()) && iEntity != null && !iEntity.equals(mc.getThePlayer()) && (targetInvisible || !iEntity.isInvisible())) {
            if (targetPlayer && classProvider.isEntityPlayer(iEntity)) {
                IEntityPlayer iEntityPlayer = iEntity.asEntityPlayer();
                if (bl) {
                    if (AntiBot.isBot(iEntityPlayer)) {
                        return false;
                    }
                    if (iEntityPlayer != false && !LiquidBounce.moduleManager.getModule(NoFriends.class).getState()) {
                        return false;
                    }
                    if (iEntityPlayer.isSpectator()) {
                        return false;
                    }
                    Teams teams = (Teams)LiquidBounce.moduleManager.getModule(Teams.class);
                    return !teams.getState() || !teams.isInYourTeam(iEntityPlayer);
                }
                return true;
            }
            return targetMobs && iEntity != false || targetAnimals && iEntity == false;
        }
        return false;
    }
}

File: sources/Kevin Reborn/kevin/utils/EntityUtils.kt
/*
 * This program is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation, either version 3 of the License, or
 *   (at your option) any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package kevin.utils

import kevin.main.KevinClient
import kevin.module.modules.misc.AntiBot.isBot
import kevin.module.modules.misc.AntiShop
import kevin.module.modules.misc.Teams
import net.minecraft.entity.Entity
import net.minecraft.entity.EntityLivingBase
import net.minecraft.entity.player.EntityPlayer


object EntityUtils : MinecraftInstance() {

    @JvmField
    var targetInvisible = true

    @JvmField
    var targetPlayer = true

    @JvmField
    var targetMobs = true

    @JvmField
    var targetAnimals = false

    @JvmField
    var targetDeath = false

    @JvmStatic
    fun isSelected(entity: Entity?, canAttackCheck: Boolean): Boolean {
        if (entity is EntityLivingBase && (targetDeath || entity.isEntityAlive) && entity != mc.thePlayer) {
            if (targetInvisible || !entity.isInvisible) {
                if (targetPlayer && entity is EntityPlayer) {

                    if (canAttackCheck) {
                        if (isBot(entity))
                            return false

                        if (entity.isClientFriend() /***&& !LiquidBounce.moduleManager.getModule(NoFriends::class.java).state***/ )
                            return false

                        if (entity.isSpectator)
                            return false

                        val antiShop = KevinClient.moduleManager.getModule(AntiShop::class.java)
                        if (antiShop.isShop(entity))
                            return false

                        val teams = KevinClient.moduleManager.getModule(Teams::class.java)
                        return !teams.state || !teams.isInYourTeam(entity)
                    }
                    return true
                }

                return targetMobs && entity.isMob() || targetAnimals && entity.isAnimal()
            }
        }
        return false
    }
    fun getPing(entityPlayer: EntityPlayer?): Int {
        if (entityPlayer == null) return 0
        val networkPlayerInfo = mc.netHandler.getPlayerInfo(entityPlayer.uniqueID)
        return networkPlayerInfo?.responseTime ?: 0
    }
}